stripe/src/main/java/com/stripe/android/paymentsheet/PaymentOptionsViewModel.kt

```diff
@@ -21,11 +21,18 @@ internal class PaymentOptionsViewModel(
     config = args.config,
     prefsRepository = prefsRepository
 ) {
+    /* TODO: Resolve userSelection, newCard, and _selection in SheetViewModel */
     private val _userSelection = MutableLiveData<PaymentSelection>()
     val userSelection: LiveData<PaymentSelection> = _userSelection
 
     override val newCard = args.newCard
 
+    // This is used in the case where the last card ws new and not saved.  In this scenario
+    // when the payment options is opened it should jump to the add card, but if the user
+    // presses the back button, they shouldn't transition to it again
+    private var transitionedOnceToAddCard = false
+    private var shouldTransitionToAddCard = newCard != null && !newCard.shouldSavePaymentMethod && !transitionedOnceToAddCard
```

Comment:
I don't think this will work the way you want it to. The expression will be evaluated once at construction time and set on `shouldTransitionToAddCard`. Changes to `transitionedOnceToAddCard` won't be reflected. Also, it appears that `shouldTransitionToAddCard` can be a `val` instead of a `var`.

You can change the statement to:
```
private val shouldTransitionToAddCard: Boolean
  get() = newCard != null && !newCard.shouldSavePaymentMethod && !transitionedOnceToAddCard
```

Which will evaluate the expression every time the property is accessed.